File: Batalla_Poo_Ruby/avion.rb
require_relative "participante.rb"

class Avion < Participante

    def getTipo
        return "Avion"
    end

    def disparar(enemigo)
       enemigo.danio(enemigo.danioParaAvion())
    end

    def danioParaTanque()
        return 200
    end

    def danioParaSoldado()
        return 500
    end

    def danioParaAvion()
        return 650
    end

end

File: Batalla_Poo_CPP/src/Avion.cpp
#include "Avion.h"


Avion::Avion():Participante()
{
    this->setNombre("Avion");
}

/*
void Avion::disparar(Participante enemigo)
{
    enemigo.quitarVida(enemigo.danioParaAvion());
}
*/

int Avion::danioParaTanque()
{
    return 200;
}

int Avion::danioParaAvion()
{
    return 500;
}

int Avion::danioParaSoldado()
{
    return 650;
}
